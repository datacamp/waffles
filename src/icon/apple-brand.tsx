import Icon from './icon-base';

type AppleBrandProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function AppleBrand({ size, ...restProps }: AppleBrandProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M2.41 6.617c.8-1.392 2.25-2.27 3.787-2.293h.047c.66 0 1.284.247 1.835.465.412.164.767.305 1.04.305.245 0 .599-.14 1.008-.3.593-.235 1.33-.526 2.102-.526.099 0 .197.005.292.014.656.03 2.285.263 3.349 1.82a.147.147 0 0 1-.042.206l-.015.01c-.31.19-1.85 1.244-1.83 3.235.022 2.451 2.049 3.35 2.28 3.445l.01.004c.07.032.103.11.08.181l-.006.016a9.93 9.93 0 0 1-1.24 2.538c-.71 1.038-1.515 2.215-2.827 2.24-.614.011-1.03-.17-1.432-.344-.41-.177-.831-.36-1.494-.36-.696 0-1.14.19-1.57.373-.382.162-.777.33-1.325.353L6.393 18c-1.165 0-2.015-1.09-2.855-2.305-1.683-2.43-2.65-6.437-1.13-9.078h.002ZM12.52 0a.148.148 0 0 1 .154.131c.108.984-.25 2.082-.954 2.938-.67.81-1.651 1.313-2.561 1.313-.064 0-.128-.002-.19-.007a.147.147 0 0 1-.136-.128c-.145-1.104.408-2.198.975-2.855C10.476.61 11.592.038 12.522 0h-.001Z"
      />
    </Icon>
  );
}

export default AppleBrand;
