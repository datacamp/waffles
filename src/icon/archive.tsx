import Icon from './icon-base';

type ArchiveProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function Archive({ size, ...restProps }: ArchiveProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M8.042 10.581 8.04 1a1 1 0 1 1 2 0l.002 9.593 2.296-2.287a.998.998 0 0 1 1.413.003c.39.392.388 1.025-.003 1.415l-3.949 3.932a.998.998 0 0 1-1.526-.014L4.335 9.708a1 1 0 1 1 1.416-1.415l2.29 2.288zM13.572 6a1 1 0 1 1 0-2H17a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H1a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1h3.41a1 1 0 1 1 0 2H2v10h14V6h-2.428z"
      />
    </Icon>
  );
}

export default Archive;
