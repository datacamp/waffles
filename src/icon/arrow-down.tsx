import Icon from './icon-base';

type ArrowDownProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function ArrowDown({ size, ...restProps }: ArrowDownProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M8 13.58 7.998 2a1 1 0 0 1 2 0L10 13.59l3.296-3.285a.998.998 0 1 1 1.41 1.417l-4.948 4.93a.998.998 0 0 1-1.527-.012l-4.938-4.933a1 1 0 0 1 0-1.414 1.002 1.002 0 0 1 1.416 0l3.29 3.287z"
      />
    </Icon>
  );
}

export default ArrowDown;
