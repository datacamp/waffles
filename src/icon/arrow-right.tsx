import Icon from './icon-base';

type ArrowRightProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function ArrowRight({ size, ...restProps }: ArrowRightProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M13.58 10 2 10.002a1 1 0 0 1 0-2L13.59 8l-3.285-3.296a.998.998 0 1 1 1.417-1.41l4.93 4.948a.998.998 0 0 1-.012 1.527l-4.933 4.938a1 1 0 0 1-1.414 0 1.002 1.002 0 0 1 0-1.416l3.287-3.29z"
      />
    </Icon>
  );
}

export default ArrowRight;
