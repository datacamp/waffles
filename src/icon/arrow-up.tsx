import Icon from './icon-base';

type ArrowUpProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function ArrowUp({ size, ...restProps }: ArrowUpProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M10 4.42 10.002 16a1 1 0 0 1-2 0L8 4.41 4.704 7.696a.998.998 0 1 1-1.41-1.417l4.948-4.93a.998.998 0 0 1 1.527.012l4.938 4.933a1 1 0 0 1 0 1.414c-.391.39-1.025.39-1.416 0l-3.29-3.287Z"
      />
    </Icon>
  );
}

export default ArrowUp;
