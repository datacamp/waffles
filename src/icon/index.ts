export { default as AddCircleInverted } from './output/add-circle-inverted';
export { default as AddCircle } from './output/add-circle';
export { default as AddUser } from './output/add-user';
export { default as AppleBrand } from './output/apple-brand';
export { default as Apply } from './output/apply';
export { default as Archive } from './output/archive';
export { default as ArrowDown } from './output/arrow-down';
export { default as ArrowLeft } from './output/arrow-left';
export { default as ArrowRight } from './output/arrow-right';
export { default as ArrowUp } from './output/arrow-up';
export { default as Assess } from './output/assess';
export { default as AssignmentInverted } from './output/assignment-inverted';
export { default as Assignment } from './output/assignment';
export { default as Attachment } from './output/attachment';
export { default as AttentionCircleInverted } from './output/attention-circle-inverted';
export { default as AttentionCircle } from './output/attention-circle';
export { default as Back } from './output/back';
export { default as BarChart } from './output/bar-chart';
export { default as Bell } from './output/bell';
export { default as Book } from './output/book';
export { default as BookmarkInverted } from './output/bookmark-inverted';
export { default as Bookmark } from './output/bookmark';
export { default as Branch } from './output/branch';
export { default as Briefcase } from './output/briefcase';
export { default as Building } from './output/building';
export { default as Bulb } from './output/bulb';
export { default as Bullet } from './output/bullet';
export { default as Calendar } from './output/calendar';
export { default as Camera } from './output/camera';
export { default as ChatInverted } from './output/chat-inverted';
export { default as Chat } from './output/chat';
export { default as CheckmarkCircleInverted } from './output/checkmark-circle-inverted';
export { default as CheckmarkCircle } from './output/checkmark-circle';
export { default as Checkmark } from './output/checkmark';
export { default as ChevronDown } from './output/chevron-down';
export { default as ChevronLeft } from './output/chevron-left';
export { default as ChevronRight } from './output/chevron-right';
export { default as ChevronUp } from './output/chevron-up';
export { default as CircleInverted } from './output/circle-inverted';
export { default as Circle } from './output/circle';
export { default as ClockInverted } from './output/clock-inverted';
export { default as Clock } from './output/clock';
export { default as Code } from './output/code';
export { default as CogInverted } from './output/cog-inverted';
export { default as Cog } from './output/cog';
export { default as Columns } from './output/columns';
export { default as Command } from './output/command';
export { default as Compass } from './output/compass';
export { default as Copy } from './output/copy';
export { default as CreditCard } from './output/credit-card';
export { default as CrossCircleInverted } from './output/cross-circle-inverted';
export { default as CrossCircle } from './output/cross-circle';
export { default as Cross } from './output/cross';
export { default as Crosshair } from './output/crosshair';
export { default as DataCampBrand } from './output/data-camp-brand';
export { default as Database } from './output/database';
export { default as Desktop } from './output/desktop';
export { default as Diamond } from './output/diamond';
export { default as Document } from './output/document';
export { default as Dollar } from './output/dollar';
export { default as Download } from './output/download';
export { default as Edit } from './output/edit';
export { default as Education } from './output/education';
export { default as ExerciseInverted } from './output/exercise-inverted';
export { default as Exercise } from './output/exercise';
export { default as Exit } from './output/exit';
export { default as Expand } from './output/expand';
export { default as ExternalLink } from './output/external-link';
export { default as FacebookBrand } from './output/facebook-brand';
export { default as Feed } from './output/feed';
export { default as Filter } from './output/filter';
export { default as Flag } from './output/flag';
export { default as FlashInverted } from './output/flash-inverted';
export { default as Flash } from './output/flash';
export { default as Flashcard } from './output/flashcard';
export { default as Float } from './output/float';
export { default as Folder } from './output/folder';
export { default as Footer } from './output/footer';
export { default as Forbidden } from './output/forbidden';
export { default as Forward } from './output/forward';
export { default as FullScreen } from './output/full-screen';
export { default as GithubBrand } from './output/github-brand';
export { default as Globe } from './output/globe';
export { default as GoogleBrand } from './output/google-brand';
export { default as GoogleColoredBrand } from './output/google-colored-brand';
export { default as GooglePlayBrand } from './output/google-play-brand';
export { default as GroupInverted } from './output/group-inverted';
export { default as Group } from './output/group';
export { default as Header } from './output/header';
export { default as HealthInverted } from './output/health-inverted';
export { default as Health } from './output/health';
export { default as HeartInverted } from './output/heart-inverted';
export { default as Heart } from './output/heart';
export { default as HelpCircleInverted } from './output/help-circle-inverted';
export { default as HelpCircle } from './output/help-circle';
export { default as Help } from './output/help';
export { default as HiddenInverted } from './output/hidden-inverted';
export { default as Hidden } from './output/hidden';
export { default as Home } from './output/home';
export { default as Image } from './output/image';
export { default as InfoCircleInverted } from './output/info-circle-inverted';
export { default as InfoCircle } from './output/info-circle';
export { default as InstagramBrand } from './output/instagram-brand';
export { default as Integration } from './output/integration';
export { default as Interface } from './output/interface';
export { default as Introduction } from './output/introduction';
export { default as KaggleBrand } from './output/kaggle-brand';
export { default as Keyboard } from './output/keyboard';
export { default as Lab } from './output/lab';
export { default as LeanpubBrand } from './output/leanpub-brand';
export { default as Learn } from './output/learn';
export { default as LineChart } from './output/line-chart';
export { default as Link } from './output/link';
export { default as LinkedInBrand } from './output/linked-in-brand';
export { default as List } from './output/list';
export { default as LockedInverted } from './output/locked-inverted';
export { default as Locked } from './output/locked';
export { default as Mail } from './output/mail';
export { default as Markdown } from './output/markdown';
export { default as Matrix } from './output/matrix';
export { default as Medal } from './output/medal';
export { default as Menu } from './output/menu';
export { default as MicrophoneInverted } from './output/microphone-inverted';
export { default as Microphone } from './output/microphone';
export { default as Mobile } from './output/mobile';
export { default as MoonInverted } from './output/moon-inverted';
export { default as Moon } from './output/moon';
export { default as More } from './output/more';
export { default as MuteInverted } from './output/mute-inverted';
export { default as Mute } from './output/mute';
export { default as NineDots } from './output/nine-dots';
export { default as Pause } from './output/pause';
export { default as PaypalBrand } from './output/paypal-brand';
export { default as PaypalColoredBrand } from './output/paypal-colored-brand';
export { default as Phone } from './output/phone';
export { default as PieChart } from './output/pie-chart';
export { default as PinInverted } from './output/pin-inverted';
export { default as Pin } from './output/pin';
export { default as PlatformSwitch } from './output/platform-switch';
export { default as PlayInverted } from './output/play-inverted';
export { default as Play } from './output/play';
export { default as Plus } from './output/plus';
export { default as Practice } from './output/practice';
export { default as Publish } from './output/publish';
export { default as PythonBrand } from './output/python-brand';
export { default as RBrand } from './output/r-brand';
export { default as Random } from './output/random';
export { default as Redo } from './output/redo';
export { default as Refresh } from './output/refresh';
export { default as RemoveInverted } from './output/remove-inverted';
export { default as Remove } from './output/remove';
export { default as Reply } from './output/reply';
export { default as RewindInverted } from './output/rewind-inverted';
export { default as Rewind } from './output/rewind';
export { default as RocketInverted } from './output/rocket-inverted';
export { default as Rocket } from './output/rocket';
export { default as Rows } from './output/rows';
export { default as Save } from './output/save';
export { default as ScalaBrand } from './output/scala-brand';
export { default as Search } from './output/search';
export { default as Share } from './output/share';
export { default as SkipInverted } from './output/skip-inverted';
export { default as Skip } from './output/skip';
export { default as SlackBrand } from './output/slack-brand';
export { default as SlackColoredBrand } from './output/slack-colored-brand';
export { default as SoundInverted } from './output/sound-inverted';
export { default as Sound } from './output/sound';
export { default as SparklesInverted } from './output/sparkles-inverted';
export { default as Sparkles } from './output/sparkles';
export { default as StarInverted } from './output/star-inverted';
export { default as Star } from './output/star';
export { default as StopInverted } from './output/stop-inverted';
export { default as Stop } from './output/stop';
export { default as SunInverted } from './output/sun-inverted';
export { default as Sun } from './output/sun';
export { default as Tab } from './output/tab';
export { default as Tag } from './output/tag';
export { default as Team } from './output/team';
export { default as TerminalInverted } from './output/terminal-inverted';
export { default as Terminal } from './output/terminal';
export { default as Text } from './output/text';
export { default as TrashInverted } from './output/trash-inverted';
export { default as Trash } from './output/trash';
export { default as TrimBegin } from './output/trim-begin';
export { default as TrimEnd } from './output/trim-end';
export { default as Trophy } from './output/trophy';
export { default as TwitterBrand } from './output/twitter-brand';
export { default as Undo } from './output/undo';
export { default as Unlink } from './output/unlink';
export { default as UnlockedInverted } from './output/unlocked-inverted';
export { default as Unlocked } from './output/unlocked';
export { default as Upload } from './output/upload';
export { default as UpvoteInverted } from './output/upvote-inverted';
export { default as Upvote } from './output/upvote';
export { default as UserInverted } from './output/user-inverted';
export { default as User } from './output/user';
export { default as VerifiedInverted } from './output/verified-inverted';
export { default as Verified } from './output/verified';
export { default as VideoCameraInverted } from './output/video-camera-inverted';
export { default as VideoCamera } from './output/video-camera';
export { default as VideoInverted } from './output/video-inverted';
export { default as Video } from './output/video';
export { default as VisibleInverted } from './output/visible-inverted';
export { default as Visible } from './output/visible';
export { default as Waypoint } from './output/waypoint';
export { default as YoutubeBrand } from './output/youtube-brand';
