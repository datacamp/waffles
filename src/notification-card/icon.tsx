import { iconWrapperStyle, iconStyle, iconBackgroundStyle } from './styles';
import NotificationCard from './notification-card';

import { tokens } from '../tokens';
import {
  CheckmarkCircleInverted,
  CrossCircleInverted,
  AttentionCircleInverted,
  InfoCircleInverted,
  RocketInverted,
} from '../icon';

const regularVariantMap = {
  default: {
    icon: <InfoCircleInverted />,
    iconColor: tokens.colors.blueDark,
    backgroundColor: tokens.colors.white,
  },
  success: {
    icon: <CheckmarkCircleInverted />,
    iconColor: tokens.colors.green,
    backgroundColor: tokens.colors.navy,
  },
  warning: {
    icon: <AttentionCircleInverted />,
    iconColor: tokens.colors.orangeLight,
    backgroundColor: tokens.colors.navy,
  },
  error: {
    icon: <CrossCircleInverted />,
    iconColor: tokens.colors.red,
    backgroundColor: tokens.colors.white,
  },
  upgrade: {
    icon: <RocketInverted />,
    iconColor: tokens.colors.purple,
    backgroundColor: 'transparent',
  },
};

const invertedVariantMap = {
  ...regularVariantMap,
  upgrade: {
    ...regularVariantMap.upgrade,
    iconColor: tokens.colors.purpleLight,
  },
};

type IconProps = {
  variant: NonNullable<
    React.ComponentProps<typeof NotificationCard>['variant']
  >;
  inverted: boolean;
};

function Icon({ variant, inverted }: IconProps) {
  const variantMap = inverted ? invertedVariantMap : regularVariantMap;

  return (
    <div css={iconWrapperStyle()}>
      <div css={iconStyle({ iconColor: variantMap[variant].iconColor })}>
        {variantMap[variant].icon}
      </div>
      <div
        css={iconBackgroundStyle({
          backgroundColor: variantMap[variant].backgroundColor,
        })}
      />
    </div>
  );
}

export default Icon;
