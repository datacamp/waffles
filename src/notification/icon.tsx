import { tokens } from '../tokens';
import {
  CheckmarkCircleInverted,
  CrossCircleInverted,
  AttentionCircleInverted,
  InfoCircleInverted,
} from '../icon';
import Notification from './notification';
import { iconStyle, iconWrapperStyle, iconBackgroundStyle } from './styles';

const variantMap = {
  success: {
    icon: <CheckmarkCircleInverted />,
    iconColor: tokens.colors.blue,
    backgroundColor: tokens.colors.navy,
  },
  error: {
    icon: <CrossCircleInverted />,
    iconColor: tokens.colors.blue,
    backgroundColor: tokens.colors.white,
  },
  warning: {
    icon: <AttentionCircleInverted />,
    iconColor: tokens.colors.blue,
    backgroundColor: tokens.colors.navy,
  },
  default: {
    icon: <InfoCircleInverted />,
    iconColor: tokens.colors.blueDark,
    backgroundColor: tokens.colors.white,
  },
};

type IconProps = {
  variant: NonNullable<React.ComponentProps<typeof Notification>['variant']>;
};

function Icon({ variant }: IconProps) {
  return (
    <div css={iconStyle()}>
      <div css={iconWrapperStyle({ iconColor: variantMap[variant].iconColor })}>
        {variantMap[variant].icon}
      </div>
      <div
        css={iconBackgroundStyle({
          backgroundColor: variantMap[variant].backgroundColor,
        })}
      />
    </div>
  );
}

export default Icon;
