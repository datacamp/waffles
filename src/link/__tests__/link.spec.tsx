import React, { useRef, useEffect } from 'react';
import { render, fireEvent } from '@testing-library/react';

import { AddCircle, ExternalLink } from '../../icon';
import { Link } from '../index';

type TestComponentProps = {
  to: string;
} & React.ComponentPropsWithoutRef<'a'>;

function TestComponent({ children, to, ...restProps }: TestComponentProps) {
  return (
    <a href={to} {...restProps}>
      {children}
    </a>
  );
}

function TestRefLink() {
  const linkRef = useRef<HTMLAnchorElement>(null);

  useEffect(() => {
    linkRef.current?.focus();
  }, []);

  return (
    <Link href="https://datacamp.com" ref={linkRef}>
      Focused With Ref
    </Link>
  );
}

describe('Link', () => {
  it('renders a link containing the text', () => {
    const { container } = render(
      <Link href="https://taylor-swift-fanclub.com">
        Join Taylor Swift Fanclub
      </Link>,
    );

    const link = container.querySelector('a');

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', 'https://taylor-swift-fanclub.com');
    expect(link).toHaveTextContent('Join Taylor Swift Fanclub');
  });

  it('sets the data attribute on the link', () => {
    const { getByTestId } = render(
      <Link data-testid="taylors-link">Join Taylor Swift Fanclub</Link>,
    );

    const link = getByTestId('taylors-link');

    expect(link).toBeInTheDocument();
  });

  it('renders left icon', () => {
    const { container, getByText } = render(
      <Link iconLeft={<AddCircle />}>Go to Fanpage</Link>,
    );

    const link = getByText('Go to Fanpage').closest('a');
    const icon = container.querySelector('svg');

    expect(link).toBeInTheDocument();
    expect(link).toContainElement(icon);
  });

  it('renders right icon', () => {
    const { container, getByText } = render(
      <Link iconRight={<ExternalLink />}>Read More</Link>,
    );

    const link = getByText('Read More').closest('a');
    const icon = container.querySelector('svg');

    expect(link).toBeInTheDocument();
    expect(link).toContainElement(icon);
  });

  it('renders both left and right icons', () => {
    const { container, getByText } = render(
      <Link iconLeft={<AddCircle />} iconRight={<ExternalLink />}>
        Go to Another Fanpage
      </Link>,
    );

    const link = getByText('Go to Another Fanpage');
    const icons = container.querySelectorAll('svg');

    expect(link).toBeInTheDocument();
    expect(icons).toHaveLength(2);
  });

  it('renders icon of custom size', () => {
    const { container, getByText } = render(
      <Link iconRight={<ExternalLink size="xsmall" />}>
        Go to Another Fanpage
      </Link>,
    );

    const link = getByText('Go to Another Fanpage').closest('a');
    const icon = container.querySelector('svg');

    expect(link).toBeInTheDocument();
    expect(link).toContainElement(icon);
    expect(icon).toHaveAttribute('width', '12');
    expect(icon).toHaveAttribute('height', '12');
  });

  it('requires props owned by custom component passed into "as" prop', () => {
    const { container } = render(
      <Link to="http://fanpage.com" as={TestComponent}>
        As a Custom Component
      </Link>,
    );

    const customComponent = container.querySelector('a');

    expect(customComponent).toBeInTheDocument();
    expect(customComponent).toHaveTextContent('As a Custom Component');
    expect(customComponent).toHaveAttribute('href', 'http://fanpage.com');
  });

  it('renders snapshot of focused state', () => {
    const { getByText } = render(<Link href="https://datacamp.com">Test</Link>);

    const link = getByText('Test').closest('a') as HTMLAnchorElement;
    fireEvent.focus(link);

    expect(link).toMatchSnapshot();
  });

  it('accepts ref and could be focused programmatically when link has a "href" attribute', () => {
    const { container } = render(<TestRefLink />);

    const link = container.querySelector('a');

    expect(link).toHaveFocus();
  });

  describe('renders snapshot', () => {
    const { container } = render(
      <Link href="https://taylor-swift-fanclub.com">
        Join Taylor Swift Fanclub
      </Link>,
    );

    const link = container.querySelector('a');

    expect(link).toMatchSnapshot();
  });

  describe('renders snapshot of inverted variant', () => {
    const { container } = render(
      <Link inverted href="https://taylor-swift-fanclub.com">
        Join Taylor Swift Fanclub
      </Link>,
    );

    const link = container.querySelector('a');

    expect(link).toMatchSnapshot();
  });
});
