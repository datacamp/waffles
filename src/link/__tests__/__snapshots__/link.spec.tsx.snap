// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  vertical-align: bottom;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  line-height: inherit;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  outline: 0;
  border-radius: 4px;
  -webkit-transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  color: #0075AD;
}

.emotion-0 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-top: -1px;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.1);
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: 0 0 0 2px rgba(5, 25, 45, 0.1);
}

<a
  class="emotion-0"
  href="https://taylor-swift-fanclub.com"
>
  <span
    class="emotion-1"
  >
    Join Taylor Swift Fanclub
  </span>
</a>
`;

exports[` 2`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  vertical-align: bottom;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  line-height: inherit;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  outline: 0;
  border-radius: 4px;
  -webkit-transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  color: #06BDFC;
}

.emotion-0 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-top: -1px;
}

.emotion-0:hover {
  background-color: rgba(255, 255, 255, 0.1);
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: 0 0 0 2px rgba(255, 255, 255, 0.1);
}

<a
  class="emotion-0"
  href="https://taylor-swift-fanclub.com"
>
  <span
    class="emotion-1"
  >
    Join Taylor Swift Fanclub
  </span>
</a>
`;

exports[`Link renders snapshot of focused state 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  vertical-align: bottom;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  line-height: inherit;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  outline: 0;
  border-radius: 4px;
  -webkit-transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  transition: box-shadow 125ms ease-out,background-color 125ms ease-out;
  color: #0075AD;
  box-shadow: 0 0 0 2px #009BD8;
}

.emotion-0 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-top: -1px;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.1);
}

<a
  class="emotion-0"
  href="https://datacamp.com"
>
  <span
    class="emotion-1"
  >
    Test
  </span>
</a>
`;
